Sources/StreamVideo/CallState.swift

```diff
@@ -180,5 +187,38 @@ public class CallState: ObservableObject {
         }
         self.ownCapabilities = event.ownCapabilities
     }
-    
+
+    private func didUpdate(_ participants: [CallParticipant]) {
+        var remoteParticipants: [CallParticipant] = []
+        var activeSpeakers: [CallParticipant] = []
+        var screenSharingSession: ScreenSharingSession?
+
+        for participant in participants {
+            if participant.id == streamVideo.user.id {
```

Comment:
This is not good enough condition, since a participant with the same id can appear from different devices. We should change the right part of the equation to the session id.